examples/nomad-consul-ami/aws_credentials

```diff
@@ -0,0 +1,3 @@
+[default]
+aws_access_key_id = TODO_ADD_YOUR_KEY_HERE
+aws_secret_access_key = TODO_ADD_YOUR_SECRET_HERE
```

Comment:
I know this is just example code, but I don't want to encourage users storing their secrets in plain text and copying them to a server. It's not a good pattern to follow. A better pattern to follow is to create an IAM role with ECR permissions and to attach that IAM role to the server (I believe the ECR cred helper supports this just fine).

Reply:
Indeed this is not the recommended production safe way. This approach was chosen for simplicity.

To adapt this to more secure way this will take some more time to update the approach and an increased configuration effort for the user. For testing nomad it might be acceptable. But nevertheless this security hole should be mentioned in the documentation.

Reply:
> To adapt this to more secure way this will take some more time to update the approach and an increased configuration effort for the user

You can make it part of the example code directly! You just attach ECR permissions to the IAM role of the Nomad client servers. It's actually much easier for the user this way, and there is no security hole.

Reply:
What do you think about adding this module to do the trick?
https://github.com/cloudposse/terraform-aws-ecr

Reply:
Interesting find. It seems to do a bit more than we need though. All we really want is an IAM role to read from ECR, right?

Reply:
It depends on the scope and the predefined environment. Since this terraform module does not provide an AWS ECR preconfiguration the user is in charge to provide or create one. When you start from scratch it could be convenient to include the batteries, but it will also enlarge the example.

Maybe it is the best to remove the AWS ECR support from this PR and create a second PR. Most important at the moment is the docker support.

Reply:
Sure, handling ECR in a separate PR makes sense.